### PR TITLE
Feat/32 check branch protection rulesets

### PR DESCRIPTION
**Pull request by AlexAxthelm**
For discussion in Tech Review this week (@hodie @RMI-PACTA/developers )

This PR adds an action to check for the existence of GitHub Rulesets, as well as some sample rulesets (in the `rulesets/` directory)

Rulesets are the successor to "Branch Protection Rules" and "Required Checks". For our purposes, the most important part is that "Branch Protection Rules" do not apply to repo admins (by default: whoever created the repo), which rulesets do.

The workflow added here is intended to be called from another repo, and by default, it will check that the "Protect Main" rule is in effect, which:
* Disallows deletion of `main`
* Disallows force pushes to `main`
* Disallows creating `main`
* Disallows pushing to `main`
* Requires Pull Requests to `main` have at least 1 approval, requires codeowner approval, dismisses stale reviews when new commits are added, and requires approval from someone other than the last committer to approve. (all roughly what we have now)

There is another ruleset example (not enabled by default), which requires certain R status checks to pass before the PR is eligible to merge.

### CODE REVIEW

File: .github/workflows/general-check-rulesets.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+---
+name: Check Rulesets
+
+on:
+  workflow_call:
+    inputs:
+      rulesets-repo:
+        description: 'Repository containing ruleset definitions to check'
+        required: false
+        default: 'RMI-PACTA/actions'
```
Comment (cjyetman) [inline .github/workflows/general-check-rulesets.yml:10]: not really sure, but seems like the "default" here is not really useful (other than being a valid possibility), so maybe "required" should be `true`?

Comment (cjyetman) [reply .github/workflows/general-check-rulesets.yml:10]: ah wait, maybe I don't understand correctly. I thought each repo would define its own rulesets, and then they could call this action from an action in their repo to verify that the appropriate rulesets in their repo are defined and configured properly.... but maybe I'm wrong?

**Review by cjyetman — APPROVED** [review]
I don't see anything glaringly, syntactically wrong here, but I think I don't fully understand how it works, so don't have much to say about it. If this is something that repo owners would implement in their repos by calling this action from an action in their repo, then maybe an example action (like other in this repo have) would be appropriate?

File: .github/workflows/general-check-rulesets.yml (hunk 1 above)
Comment (AlexAxthelm) [reply .github/workflows/general-check-rulesets.yml:10]: weird part about GH Actions: if `required: true`, then the workflow fails if the input is not explicitly included with `uses:` in the calling workflow (Thereby nullifying the value of having a default ¯\_(ツ)_/¯)

I'm inclined to keep this as a input that we don't expect to use often, but it is possible to define rulesets in other repos if we want.

**Comment by AlexAxthelm** [comment]
> then maybe an example action (like other in this repo have) would be appropriate?

Yep. Planning on getting a number of the general/housekeeping checks ready together soon in a default/example action.

**Review by jdhoffa — COMMENTED** [review]
retroactively lgtm :-)

**Comment by MonikaFu** [comment]
does this require an action from me as a maintainer?

**Comment by jdhoffa** [comment]
> does this require an action from me as a maintainer?

No action required.

At this stage all actions in the [actions repo](https://github.com/RMI-PACTA/actions/) are optional. This may change, however not before long drawn-out discussions with the Tech Team. 

Certain actions might be useful to you (e.g. https://github.com/RMI-PACTA/actions/blob/main/examples/R.yml and https://github.com/RMI-PACTA/actions/blob/main/examples/issues.yml) 

but they are entirely at the discretion of the maintainer.

**Pull request merged**